packages/shared/src/components/cores/CoreOptionList.tsx

```diff
@@ -1,17 +1,40 @@
 import type { ReactElement } from 'react';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import {
   CoreOptionButton,
   CoreOptionButtonPlaceholder,
 } from './CoreOptionButton';
 import { useProductPricing } from '../../hooks/useProductPricing';
+import { promisifyEventListener } from '../../lib/func';
+import { stringToBoolean } from '../../lib/utils';
+import { useBuyCoresContext } from '../../contexts/BuyCoresContext/types';
 import { PurchaseType } from '../../graphql/paddle';
 
 export const CoreOptionList = (): ReactElement => {
+  const { selectedProduct } = useBuyCoresContext();
+  const [isLoadingNative, setLoadingNative] = useState(false);
   const { data: prices, isPending: isPendingPrices } = useProductPricing({
     type: PurchaseType.Cores,
   });
 
+  useEffect(() => {
+    promisifyEventListener<void, 'true' | 'false'>(
+      'iap-loading',
```

Comment:
Is this some native thing?

Reply:
yes, just while stuff is loading on native side, using it for plus as well